### PR TITLE
Bump SDK

### LINKED ISSUE #7: aws-sdk v2?
Hey

Is there any plan to bump it to use the aws-sdk v2?

### PR DESCRIPTION
**Pull request by JDutil**
We're interested in using this library with the latest sdk so this is the current WIP for our updates.

I've also got a branch containing a bunch of syntax updates I made while reading the code to fix rubocop warnings that I'd like to contribute, but am trying to keep this PR's changes minimal to just what is required first.  You can check those out over at https://github.com/aws/aws-sessionstore-dynamodb-ruby/compare/master...JDutil:bump-aws-sdk 

fixes #7

### CODE REVIEW

**Comment by JDutil** [comment]
I've fixed the broken specs I was running into so this build should now be passing.

**Comment by phstc** [comment]
> https://github.com/aws/aws-sessionstore-dynamodb-ruby/issues/7#issuecomment-270953691

@awood45 any chance to get this merged? 🙏

**Comment by awood45** [comment]
I've just returned from vacation, I'm going to start looking at this now. This will have to be a major version bump of this Gem, so will also look to see if there are other improvements we should bundle with such a change.

File: aws-sessionstore-dynamodb.gemspec [hunk 1]
```diff
@@ -13,6 +13,6 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
-  spec.add_dependency 'aws-sdk-v1'
-  spec.add_dependency 'rack', '~> 1.0'
+  spec.add_dependency 'aws-sdk'
```
Comment (awood45) [inline aws-sessionstore-dynamodb.gemspec]: Should still keep a major version lock on this, such as `'~> 2.0'`.

Both because this in theory would allow a V1 dependency which would then mysteriously fail, and because if there's ever a V3, we don't want this to hard fail.

File: lib/aws/session_store/dynamo_db/configuration.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # language governing permissions and limitations under the License.
 
 require 'yaml'
-require 'aws-sdk-v1'
+require 'aws-sdk'
 
 module AWS::SessionStore::DynamoDB
```
Comment (awood45) [inline lib/aws/session_store/dynamo_db/configuration.rb]: Thing I'm unsure about: If we should change the top-level module to be `Aws` to match the nomenclature across other such libraries. There are valid arguments to both sides of this question.

File: lib/aws/session_store/dynamo_db/configuration.rb [hunk 3]
```diff
@@ -154,7 +153,7 @@ class Configuration
     #   See AWS DynamoDB documentation for table write_capacity for more
     #   information on this setting.
     # @option options [DynamoDB Client] :dynamo_db_client
-    #   (AWS::DynamoDB::ClientV2) DynamoDB client used to perform database
+    #   (Aws::DynamoDB::ClientV2) DynamoDB client used to perform database
```
Comment (awood45) [inline lib/aws/session_store/dynamo_db/configuration.rb]: There is no `ClientV2` in V2 of the SDK. Just `Client`.

File: lib/aws/session_store/dynamo_db/locking/base.rb [hunk 4]
```diff
@@ -56,7 +56,7 @@ def delete_session(env, sid)
     def handle_error(env = nil, &block)
       begin
         yield
-      rescue AWS::DynamoDB::Errors::Base => e
+      rescue Aws::DynamoDB::Errors::Base => e
```
Comment (awood45) [inline lib/aws/session_store/dynamo_db/locking/base.rb]: `Aws::DynamoDB::Errors::ServiceError` is the "Base" error class in V2.

File: lib/aws/session_store/dynamo_db/configuration.rb (hunk 2 above)
Comment (JDutil) [inline lib/aws/session_store/dynamo_db/configuration.rb]: @awood45 it would be nice to change the top-level module, but I'd do that in it's own PR to make code review simpler which is why I didn't bother for now.

File: lib/aws/session_store/dynamo_db/locking/base.rb (hunk 4 above)
Comment (JDutil) [inline lib/aws/session_store/dynamo_db/locking/base.rb]: @awood45 I tried changing to ServiceError, but it breaks one of the tests:
```
2.3.3 in aws-sessionstore-dynamodb-ruby/ on bump-sdk
› be rake
bundle exec rspec -t ~integration
Run options: exclude {:integration=>true}
.........F................

Failures:

  1) AWS::SessionStore::DynamoDB Error handling for Rack Middleware with default error handler catches exception for inaccurate table key
     Failure/Error: lambda do
       expected #<Aws::DynamoDB::Errors::ValidationException: The provided key element does not match the schema> but nothing was raised
     # ./spec/aws/session_store/dynamo_db/error/default_error_handler_spec.rb:44:in `block (3 levels) in <top (required)>'

Finished in 0.12099 seconds
26 examples, 1 failure

Failed examples:

rspec ./spec/aws/session_store/dynamo_db/error/default_error_handler_spec.rb:41 # AWS::SessionStore::DynamoDB Error handling for Rack Middleware with default error handler catches exception for inaccurate table key
rake aborted!
Command failed with status (1): [bundle exec rspec -t ~integration...]
/Users/JDutil/Code/aws-sessionstore-dynamodb-ruby/tasks/test.rake:17:in `block (2 levels) in <top (required)>'
/Users/JDutil/.rbenv/versions/2.3.3/bin/bundle:22:in `load'
/Users/JDutil/.rbenv/versions/2.3.3/bin/bundle:22:in `<main>'
Tasks: TOP => default => test:unit
(See full trace by running task with --trace)
```

File: lib/aws/session_store/dynamo_db/configuration.rb (hunk 2 above)
Comment (awood45) [inline lib/aws/session_store/dynamo_db/configuration.rb]: That's reasonable enough. Like I said, there's probably a number of refactors we need to make or want to make before a full release of this. For example, I'm wondering if using `Aws::Record` wouldn't simplify a ton of this code, now that it exists.

File: lib/aws/session_store/dynamo_db/locking/base.rb (hunk 4 above)
Comment (JDutil) [inline lib/aws/session_store/dynamo_db/locking/base.rb]: @awood45 There was another reference to base class I needed to update, but specs are passing again.

**Comment by JDutil** [comment]
@awood45 okay initial feedback is updated. Let me know if you need anything else.

**Comment by dkoprov** [comment]
@awood45 any updates on this issue?

**Comment by awood45** [comment]
Sorry, I've been swamped lately. Also, admittedly, waiting to see if the [newest DynamoDB feature](https://aws.amazon.com/blogs/aws/new-manage-dynamodb-items-using-time-to-live-ttl/) could play a role once released.

**Comment by phstc** [comment]
@awood45 [hehe ..](https://twitter.com/pablocantero/status/836319777555693569). Is the TTL configuration already available through the SDK?

**Comment by awood45** [comment]
I will certainly update this space when it is. If we are major version bumping the session store, it may be worth using that version as the minimum, since TTL may nicely solve some performance issues in the previous iteration of this gem.

**Comment by JDutil** [comment]
Would be nice to move forward with this specific issue of updating the sdk version used, and then look at TTL after. Even if that means just updating master branch, and waiting to release the major version bump until TTL support is added. There is another open issue related to using TTL here https://github.com/aws/aws-sessionstore-dynamodb-ruby/issues/12

**Comment by awood45** [comment]
We can do that - there's also the option of a pre-release of the MV bump.

**Comment by phstc** [comment]
@awood45 I think a specific issue and PR for TTL as mentioned by @JDutil does make sense.

**Comment by cjyclaire** [comment]
Latest release contains merged PR #17 that supports V2 SDK, it covers all this PR covers as well, closing : )

**Comment by phstc** [comment]
@cjyclaire great! How about the other opened pull requests adding the `expire_at` attribute?
